### DISCUSSION TITLE
Error while compiling LDT

### DISCUSSION THREAD

**Discussion by modeler22** [NASA-LIS/LISF#1040] [Q&A]
Dear all,

I am trying to compile the LDT using intel compilers, however I am getting the following error message. Please suggest me how to overcome with this problem (I have also listed the environment variables and attached to configure.ldt file here). Thanks...

**Error message** 

mpiifort -c -O2 -u -traceback -fpe0  -nomixed_str_len_arg -names lowercase -convert big_endian -assume byterecl  -DSPMD -DIFC -I/home/usr/Software/intel_soft/esmf/mod/modO/Linux.intel.64.intelmpi.default -DUSE_INCLUDE_MPI -I/home/usr/Software/intel_soft/eccodes/include/ -I/home/usr/Software/intel_soft/netcdf/include/  -I./ -I../lib/xmrg/ -I../params/VIC/ -I../USAFSI/ ../params/JULES/JULES50_parmsMod.F90
ifort: command line warning #10434: option '-nomixed_str_len_arg' use with underscore is deprecated; use '-nomixed-str-len-arg' instead
mpiifort -c -O2 -u -traceback -fpe0  -nomixed_str_len_arg -names lowercase -convert big_endian -assume byterecl  -DSPMD -DIFC -I/home/usr/Software/intel_soft/esmf/mod/modO/Linux.intel.64.intelmpi.default -DUSE_INCLUDE_MPI -I/home/usr/Software/intel_soft/eccodes/include/ -I/home/usr/Software/intel_soft/netcdf/include/  -I./ -I../lib/xmrg/ -I../params/VIC/ -I../USAFSI/ ../core/LDT_ANNMod.F90

ifort: command line warning #10434: option '-nomixed_str_len_arg' use with underscore is deprecated; use '-nomixed-str-len-arg' instead
../core/LDT_ANNMod.F90(1069): error #6627: This is an actual argument keyword name, and not a dummy argument name.   [MODE]
            mode=NF90_CLOBBER, ncid=ftn)
------------^
../core/LDT_ANNMod.F90(1287): error #6627: This is an actual argument keyword name, and not a dummy argument name.   [MODE]
            mode=NF90_CLOBBER, ncid=ftn)
------------^
compilation aborted for ../core/LDT_ANNMod.F90 (code 1)
make: *** [LDT_ANNMod.o] Error 1

[ERR] Compile failed

**My environment variables to compile LDT are**

export LDT_ARCH=linux_ifc
export LDT_SPMD=parallel
export LDT_FC=mpiifort
export LDT_CC=mpicc

export LDT_MODESMF=/home/usr/Software/intel_soft/esmf/mod/modO/Linux.intel.64.intelmpi.default
export LDT_LIBESMF=/home/usr/Software/intel_soft/esmf/lib/libO/Linux.intel.64.intelmpi.default

export LDT_OPENJPEG=/home/usr/Software/intel_soft/openjpeg
export LDT_ECCODES=/home/usr/Software/intel_soft/eccodes
export LDT_NETCDF=/home/usr/Software/intel_soft/netcdf

export LD_LIBRARY_PATH=$LDT_OPENJPEG/lib:$LDT_LIBESMF:$LDT_NETCDF/lib:${LDT_ECCODES}/lib:$LD_LIBRARY_PATH

[configure.ldt.txt](https://github.com/NASA-LIS/LISF/files/8408949/configure.ldt.txt)

**Comment by dmocko** [comment]
Hi @modeler22 - sorry for the slow response to your question.

What version of the netCDF library are you using?  From your configure.ldt file, you likely entered "3" to the netCDF version question when running the configure script.  Can you please confirm the version you have on your system?

You may need to use a more modern version of the netCDF libraries.  The build document below says that netCDF 3.6.3 or 4.5.0 (or later) is needed.  Please note the section about building netCDF 4 with first the C libraries and then with the FORTRAN libraries:

https://github.com/NASA-LIS/LISF/blob/master/docs/LDT_users_guide/build.adoc
